### PR TITLE
Closes #188: Linting

### LINKED ISSUE #188: Add Prettier & ESLint
- [ ] Configure Prettier & ESLint incl. GitHub Actions / CI

### PR DESCRIPTION
**Pull request by bitionaire**
Added linting, in the first step auto-formatting and linting will be applied on staged files. We should roll-out this feature to the CI once we fixed a significant amount of files in order to prevent a large number of merge conflicts.

### CODE REVIEW

File: .husky/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+_
```
Comment (brandstetterm) [inline .husky/.gitignore:1]: Is that correct?

File: server/.prettierrc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "printWidth": 180,
+  "arrowParens": "always",
+  "semi": false,
```
Comment (brandstetterm) [inline server/.prettierrc:4]: Ich would prefer `semi: true`

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "printWidth": 180,
+  "arrowParens": "always",
+  "semi": false,
```
Comment (brandstetterm) [inline .prettierrc:4]: Ich would prefer `semi: true`

**Review by brandstetterm — CHANGES_REQUESTED** [review]

File: .husky/.gitignore (hunk 1 above)
Comment (bitionaire) [reply .husky/.gitignore:1]: This is auto-generated by husky and ignores `husky.sh` within the directory `_`, so seems correct for me 🤷

**Review by brandstetterm — APPROVED** [review]

**Pull request merged**